databricks_cli/dbfs/api.py

```diff
@@ -124,7 +125,17 @@ def get_file(self, dbfs_path, dst_path, overwrite, headers=None):
                 local_file.write(b64decode(data))
 
     def delete(self, dbfs_path, recursive, headers=None):
-        self.client.delete(dbfs_path.absolute_path, recursive=recursive, headers=headers)
+        while True:
+            try:
+                self.client.delete(dbfs_path.absolute_path, recursive=recursive, headers=headers)
+            except HTTPError as e:
+                # Handle partial delete exceptions and retry until all the files have been deleted
+                if (e.response.status_code == 429 and
```

Comment:
If we go for 503, is it going to be possible to tell apart 503 "service unavailable" from 503 "partial delete" here?

Reply:
No, unless we add a special field in the exception or change `error_code` to something specific to partial delete.

Reply:
Solved by using a dedicated `error_code` in the response.